### PR TITLE
feat: t5 - get all unique extensions of files

### PR DESCRIPTION
**Pull request by rtest12**
## Task 5 - get all unique extensions of files

Script should get a folder as an argument and return a list of unique extensions of all files (including in the nested directories)

name the file: `chapter2-l2-t5-get-file-unique-extensions.sh`

example of the call:

`./chapter2-l2-t5-get-file-unique-extensions.sh ~/tmp/bazel`
```
c
h
zip
```
## My output:
```
✦13 ➜ ./chapter2-l2-t5-get-file-unique-extensions.sh /home/maxim/test
pem
pub
sh
t4
txt
yaml

```

### CODE REVIEW

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# Task 5 - get all unique extensions of files
+set -eo pipefail
+
+
+# Error handling function.
+err() {
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh:10]: Please make `err` function in the same way I asked you in previous PRs, i.e. add `[ERROR]` prefix and make `exit` from it

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $*" >&2
+}
+
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh:16]: something works incorrectly here
image

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+fi
+
+# Find only regular files and put in array
+files=$(find $1 -type f)
+
+
+# Then we cut out the paths and file names, leaving only unique extensions.
+for file in $files; do
+  list=${file##*/}
+  echo ${list##*.}
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh]: always quote strings passed to `echo`

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  err "Specify a single directory."
+  exit 1
+fi
+
+# Find only regular files and put in array
+files=$(find $1 -type f)
+
+
+# Then we cut out the paths and file names, leaving only unique extensions.
+for file in $files; do
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh]: Let's follow style guide `${files}`

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  list=${file##*/}
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh:30]: let's add info that it is local var

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh (hunk 2 above)
Comment (rtest12) [reply chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh:16]: fixed

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+fi
+
+# Find only regular files and put in array
+files=$(find "$1" -type f)
+
+
+# Then we cut out the paths and file names, leaving only unique extensions.
+for file in ${files}; do
+# local var list
+  list=${file##*/}
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh:30]: It is a bad practice to call vars as vars types, i.e. list / map / string and etc. Please find more appropriate name for it

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+#
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh:2]:
```suggestion

```

File: chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+
+# Then we cut out the paths and file names, leaving only unique extensions.
+for file in ${files}; do
+# local var list
+  list=${file##*/}
+  if echo "${list}" | grep -q "\."; then
+    echo "${list##*.}"
+  fi
+done | sort -u
```
Comment (kseniyashaydurova) [inline chapter2/tests/t5/chapter2-l2-t5-get-file-unique-extensions.sh:34]: Pls add one empty line at the end

**Review by kseniyashaydurova — APPROVED** [review]

**Pull request merged**